astropy/io/ascii/latex.py

```diff
@@ -80,13 +82,21 @@ class LatexSplitter(core.BaseSplitter):
     '''
     delimiter = '&'
 
+    def __call__(self, lines):
+        last_line = RE_COMMENT.split(lines[-1])[0].strip()
+        if not last_line.endswith(r'\\'):
+            print(last_line)
+            lines[-1] = last_line + r'\\'
+
+        return super(LatexSplitter, self).__call__(lines)
+
     def process_line(self, line):
         """Remove whitespace at the beginning or end of line. Also remove
         \\ at end of line"""
-        line = line.split('%')[0]
+        line = RE_COMMENT.split(line)[0]
```

Comment:
I don't know enough about regexpes: What is the line has white space list so: `data1 & data2 \\     %and now a comment``?